Lib/ast.py

```diff
@@ -702,11 +703,20 @@ def maybe_newline(self):
         if self._source:
             self.write("\n")
 
-    def fill(self, text=""):
+    def maybe_semicolon(self):
+        """Adds a "; " delimiter if it isn't the start of generated source"""
+        if self._source:
+            self.write("; ")
+
+    def fill(self, text="", *, allow_semicolon=True):
```

Comment:
Actually, when do you need allow_semicolon = True? is there an AST class for which this would always be the case? if so, you can also maybe use a private attribute instead of an argument?

Reply:
All the non-block statements can always take semicolon, but semicolon only applies when unparsing an Interactive mod (never used in unparse otherwise). Also don't know where you would want to put it, `self` in `self.fill()` is the `_Unparser` and the actual target node is not passed. As well `self.fill()` is used for things other than ast nodes, so...?

Reply:
Ah right fill does not know the node being currently handled... ok forget my suggestion.